core/src/scene/sceneLoader.cpp

```diff
@@ -271,6 +273,156 @@ void SceneLoader::loadCameras(Node cameras, View& view) {
+
+            filter = generateNoneFilter(_filter["not"]);
+
+        } else if (filtItr->first.as<std::string>() == "any") {
+
+            filter = generateAnyFilter(_filter["any"]);
+
+        } else if (filtItr->first.as<std::string>() == "all") {
+
+            filter = generateFilter(_filter["all"]);
+
+        } else {
+
+            std::string key = filtItr->first.as<std::string>();
+            filter = generatePredicate(_filter[key], key);
+
+        }
+
+        filters.push_back(filter);
+
+    }
+
+    if(filters.size() > 0) {
+        return (new All(filters));
+    } else {
+        return nullptr;
+    }
+
+}
+
+Filter* SceneLoader::generatePredicate(YAML::Node _node, std::string _key) {
+
+    if(_node.IsScalar()) {
+        try {
+            float value = _node.as<float>();
+            return (new Equality(_key, {new NumValue(value)}));
+        } catch(const BadConversion& e) {
+            std::string value = _node.as<std::string>();
+            // NOTE: there is no way to distinguish between boolean true/false and a string "true"/"false". Yaml-cpp will consider both equivalent
+            if(value == "true") {
+                return (new Equality(_key, {new NumValue(1)}));
+            } else if(value == "false") {
+                return (new Equality(_key, {new NumValue(0)}));
+            } else {
+                return (new Equality(_key, {new StrValue(value)}));
+            }
+        }
+    } else if(_node.IsSequence()) {
+        ValueList values;
+        for(YAML::const_iterator valItr = _node.begin(); valItr != _node.end(); ++valItr) {
+            try {
+                float value = valItr->as<float>();
+                values.emplace_back(new NumValue(value));
+            } catch(const BadConversion& e) {
+                std::string value = valItr->as<std::string>();
+                // NOTE: there is no way to distinguish between boolean true/false and a string "true"/"false". Yaml-cpp will consider both equivalent
+                if(value == "true") {
+                    values.emplace_back(new NumValue(1));
+                } else if(value == "false") {
+                    values.emplace_back(new NumValue(0));
+                } else {
+                    values.emplace_back(new StrValue(value));
+                }
+            }
+        }
+        return (new Equality(_key, std::move(values)));
+    } else if(_node.IsMap()) {
+        float minVal = -std::numeric_limits<float>::infinity();
+        float maxVal = std::numeric_limits<float>::infinity();
+
+        for(YAML::const_iterator valItr = _node.begin(); valItr != _node.end(); ++valItr) {
+            if(valItr->first.as<std::string>() == "min") {
+                try {
+                    minVal = valItr->second.as<float>();
+                } catch(const BadConversion& e) {
+                    logMsg("Error: Badly formed filter.\tExpect a float value type, string found.\n");
+                    return nullptr;
```

Comment:
Not sure if `nullptr` is the best thing to return here. This means that if someone writes a malformed filter, the whole app is going to segfault when it tries to filter incoming data. It might be better to keep that error message, but return a filter that always fails (for example) so that the rest of the app can still function. We could return range(-inf, -inf), or we could create a distinct "NullFilter" that simply always returns false, or we could make `Filter` a concrete type and have it always return false.